test/unit/features/debug.spec.js

```diff
@@ -80,4 +80,38 @@ found in
            <Root>`
     ).toHaveBeenWarned()
   })
+
+  describe('warn', () => {
+    const msg = 'message'
+    const vm = new Vue()
+
+    it('calls warnHandler if warnHandler is set', () => {
+      Vue.config.warnHandler = jasmine.createSpy()
+
+      warn(msg, vm)
+
+      expect(Vue.config.warnHandler).toHaveBeenCalledWith(msg, vm, jasmine.any(String))
```

Comment:
As you can see here, I don't explicitly match the `trace` argument as I cannot spy the `generateComponentTrace` method without exporting it and wanted to avoid that without approval from a maintainer.